### PR TITLE
[lager] add port

### LINKED ISSUE #19538: [New Port Request] lager
Library name:
lager

Library description:
C++ library for value-oriented design using the unidirectional data-flow architecture

Source repository URL:
https://github.com/arximboldi/lager

Project homepage (if different from the source repository):
https://sinusoid.es/lager/

Anything else that is useful to know when adding (such as optional features the library may have that should be included):
Depends on zug (#19537). I have a PR incoming to add it, please wait.

### PR DESCRIPTION
**Pull request by waybeforenow**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #19538 

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  I've tested `x86-linux`, will wait for CI for the rest.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/lager/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+)
+
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+        docs  lager_BUILD_DOCS
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
```
Comment (JonLiu1993) [inline ports/lager/portfile.cmake:16]:
```suggestion
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
```

File: ports/lager/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    OPTIONS
+        -Dlager_BUILD_EXAMPLES=OFF
+        -Dlager_BUILD_TESTS=OFF
+        ${FEATURE_OPTIONS}
+)
+
+
+vcpkg_install_cmake()
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/Lager)
```
Comment (JonLiu1993) [inline ports/lager/portfile.cmake:25]:
```suggestion
vcpkg_cmake_install()
vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/Lager)
```

File: ports/lager/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug" "${CURRENT_PACKAGES_DIR}/lib")
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (JonLiu1993) [inline ports/lager/portfile.cmake:28]:
```suggestion
file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```

File: ports/lager/vcpkg.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "lager",
+  "version-string": "2021-06-17",
+  "port-version": 1,
```
Comment (JonLiu1993) [inline ports/lager/vcpkg.json:4]:
```suggestion
  "version-date": "2021-06-17",
```

File: ports/lager/vcpkg.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  "description": "C++ library for value-oriented design using the unidirectional data-flow architecture",
+  "homepage": "https://sinusoid.es/lager/",
+  "dependencies": [
+    "boost-hana",
+    "vcpkg-cmake-config",
+    "zug"
```
Comment (JonLiu1993) [inline ports/lager/vcpkg.json:10]:
```suggestion
    "boost-hana",
    {
      "name": "vcpkg-cmake",
      "host": true
    },
    {
      "name": "vcpkg-cmake-config",
      "host": true
    }
    "zug"
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by waybeforenow** [comment]
@JonLiu1993, I incorporated your changes and re-made the branch entirely, please check it again. Thanks

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
Feature 'docs' tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static
- x64-linux

**Review by ras0219-msft — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the new port! 👍

**Pull request merged**